### PR TITLE
App setup & Prayer times page

### PR DESCRIPTION
**Pull request by TarekkMA**
Still WIP PR.

### CODE REVIEW

**Comment by TarekkMA** [comment]
@gemy21ce I did complete the application setup and the general structure I think is now well-defined in my head,

I'm using the following packages:
 * `dio`: http client
 *  `enum_to_string` `strings`: useful utility functions
 * `hijri` convert to hijri date
 * `get_it` `injectable`: for dependency injection 
 * `riverpod`: dependency injection
 * `flutter_hooks` + `riverpod` + `state_notifier`: state management solution
 * `material_design_icons_flutter`: pretty icons
 * `pretty_dio_logger`: network logging
 * `freezed`: union/data classes
 * `json_serializable`: json serialization 


When you get the chance please review the UI, if every thing is okay I will refactor the UI code to make it cleaner and add some tests.

Screen Shot 2021-03-30 at 10 12 10 PM

File: lib/data/models/app/salawat.dart [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  FAJR,
+  ZUHR,
+  ASR,
+  MAGHRIB,
+  ISHA,
+}
+
+extension SalahExt on Salah {
+  String getStringName() {
+    return strings.capitalize(EnumToString.convertToString(this).toLowerCase());
```
Comment (gemy21ce) [inline lib/data/models/app/salawat.dart]: why to lowercase then to capitalize?
and why not to use just this.toString().toUpperCase()

Comment (TarekkMA) [reply lib/data/models/app/salawat.dart]: capitalize only make the first character upper case

**Comment by TarekkMA** [comment]
I think this PR is ready for a review

**Comment by TarekkMA** [comment]
I have added a CI check that does the following:
- run build_runner
- test
- build for android
- build for ios

**Review by gemy21ce — APPROVED** [review]

**Pull request merged**